### PR TITLE
Added MATLAB scripts and instructions

### PR DESCRIPTION
**Pull request by marianalevineu**
Discussed topics:
Example scripts to run MATLAB on Discovery. The examples include:
1. Serial (1 CPU) script
2. Parallel (multiple CPU) script
3. Parallel (multiple CPU) script with job array
4. Parallel GPU script

### CODE REVIEW

**Review by sriram94aa — APPROVED** [review]

File: MATLAB/README.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+Example scripts to run MATLAB on Discovery. The examples include:
+
+1. Serial (1 CPU) script
+2. Parallel (multiple CPU) script
+3. Parallel (multiple CPU) script with job array 
+4. Parallel GPU script
+
+## Serial example
+
+The script **submit_serial.bash** shell script is a basic example of running a serial batch job using MATLAB. It runs the MATLAB function **magic_square.m** on a single CPU (`--ntasks=1`) on the express partition.
```
Comment (ShobanaSekar) [inline MATLAB/README.md:11]: very minor but script mentioned twice in line 11

Comment (ShobanaSekar) [reply MATLAB/README.md:11]: inside the bracket after single CPU, should it be --cpus-per-task=1

File: MATLAB/README.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+```bash
+#!/bin/bash
+#SBATCH -N 1			#use one compute node
+#SBATCH --ntasks=1		#use 1 task
+##SBATCH --cpus-per-task=1	#use 1 cpu per task (would be the default)
```
Comment (ShobanaSekar) [inline MATLAB/README.md:17]: Is the double # intentional?

File: MATLAB/README.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+#SBATCH --output=test.out	#standard output file name
+#SBATCH --error=test.err	#standard error file name
+#SBATCH --job-name=test		#name for the job allocation
+
+#To gain access to Matlab, a Matlab module must be loaded:
+module load matlab/R2020a
+
+#To submit Matlab jobs to the SLURM scheduler, the Matlab commands to be executed must be containined in a single .m script:
+
+matlab -nodisplay < magic_square.m
```
Comment (ShobanaSekar) [inline MATLAB/README.md:28]: can we add a sentence on what -nodisplay is for

File: MATLAB/README.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+#To submit Matlab jobs to the SLURM scheduler, the Matlab commands to be executed must be containined in a single .m script:
+
+matlab -nodisplay < parforExample.m 
+
+# Cleanup local work directory
+rm -rf /scratch/$USER/$SLURM_JOB_ID
+
+```
+
+Note that the MATLAB script **parforExample.m** has to include MATLAB Parallel synthax to ensure parallelism is invoked:
```
Comment (ShobanaSekar) [inline MATLAB/README.md:73]: syntax typo

File: MATLAB/README.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+% run a parfor loop, distributing the iterations to the SLURM_CPUS_ON_NODE workers
+
+% 1. read the data and parameters (and iteration number N) from file: data
+
+parfor i = N:1000
+	%1. do a calculation with the data and paramters of i
+end
+
+%2. store the current data / current parameters into a file: data
```
Comment (ShobanaSekar) [inline MATLAB/README.md:95]: Is there an additional step/command here to store the data into a file?

File: MATLAB/matlab_gpu/submit_gpu.bash [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+#SBATCH -N 1			#use one compute node
+#SBATCH --ntask=1		#use 1 task
```
Comment (ShobanaSekar) [inline MATLAB/matlab_gpu/submit_gpu.bash:3]: --ntasks=1

File: MATLAB/myParallelAlgorithmFcn.m [hunk 7]
```diff
@@ -0,0 +1,27 @@
+function [numWorkers,time] = myParallelAlgorithmFcn ()
```
Comment (ShobanaSekar) [inline MATLAB/myParallelAlgorithmFcn.m:1]: is the space between myParallelAlgorithmFcn and the () expected?

**Review by ShobanaSekar — COMMENTED** [review]
Hey @marianalevineu I've added few comments, hope those make sense. Let me know if you have any questions!

**Pull request merged**